### ISSUE TITLE
feat: add Validation to/in Autocomplete control

### ISSUE THREAD

**Issue by VBTXDeveloper** [microsoft/fluentui-blazor#3115]
# 🙋 Feature Request

Using the Autocomplete control, we are dynamically setting required to "true" when a different control is populated.
This properly sets the requirement; you cannot submit the form when the control is empty. However, the field remains required and will not validate successfully, even after you select a value. It would be nice for the validation to properly accept a value and allow for submission.

This was [reported as a Bug](https://github.com/microsoft/fluentui-blazor/issues/3025), but marked 'working as designed'. So I'm converting it to a feature request for future consideration.

## 🤔 Expected Behavior

When we select a value in the autocomplete control, it should no longer say that the field is required, and allow the form to submit.

## 😯 Current Behavior

Field fails validation, even after a selection is made.
![Image](https://github.com/user-attachments/assets/948c3849-08e9-4bca-b24f-5bbd6bc558f7)

## 🔦 Context

We would like to have a an autocomplete control be required when another field is selected. The control should require a selection and validate properly.

## 💻 Examples
Examples provided in referenced Bug ticket.
![Image](https://github.com/user-attachments/assets/4f92bd80-5a09-4047-b8a2-8289b2b34d5e)

**Comment by dvoituron** [comment]
Indeed. This feature is not yet implemented, as described in the doc. 

I don't think we'll have time to develop it before v5. You have to handle this exception manually. 

![image](https://github.com/user-attachments/assets/b60ba041-c7e4-4bea-8814-d49b15faeb49)

**Comment by joriverm** [comment]
@dvoituron : i was looking at this and have implemented something in the listcomponentbase that seems to work. want me to PR this and see if it is up to spec or contains mistakes?

**Comment by joriverm** [comment]
during cleanup and testing of the code i noticed a multiple fluentselect works now but autocomplete doesnt due to the fieldidentifier being wrong. (it triggers validation, but sends SelectedOptions as field, not the bound property/model)
ill try and figure it out what is going on ! :)
